src/index.js

```diff
@@ -161,199 +134,137 @@ class IpfsRepo {
    * be returned in the callback if one has been created.
    *
    * @param {string} path
-   * @param {function(Error, lockfile)} callback
-   * @returns {void}
+   * @returns {Promise<lockfile>}
    */
-  _openLock (path, callback) {
-    this._locker.lock(path, (err, lockfile) => {
-      if (err) {
-        return callback(err, null)
-      }
-
-      assert.strictEqual(typeof lockfile.close, 'function', 'Locks must have a close method')
-      callback(null, lockfile)
-    })
+  async _openLock (path) {
+    const lockfile = await this._locker.lock(path)
+    assert.strictEqual(typeof lockfile.close, 'function', 'Locks must have a close method')
+    return lockfile
   }
 
   /**
    * Closes the lock on the repo
    *
-   * @param {function(Error)} callback
-   * @returns {void}
+   * @returns {Promise<void>}
    */
-  _closeLock (callback) {
+  async _closeLock () {
     if (this.lockfile) {
-      return this.lockfile.close(callback)
+      return this.lockfile.close()
     }
-    callback()
   }
 
   /**
    * Check if the repo is already initialized.
-   *
    * @private
-   * @param {function(Error)} callback
-   * @returns {void}
+   * @returns {Promise<bool>}
    */
-  _isInitialized (callback) {
+  async _isInitialized () {
     log('init check')
-    parallel(
-      {
-        config: (cb) => this.config.exists(cb),
-        spec: (cb) => this.spec.exists(cb),
-        version: (cb) => this.version.check(repoVersion, cb)
-      },
-      (err, res) => {
-        log('init', err, res)
-        if (err && !res.config) {
-          return callback(Object.assign(new Error('repo is not initialized yet'),
-            {
-              code: ERRORS.ERR_REPO_NOT_INITIALIZED,
-              path: this.path
-            }))
-        }
-        callback(err)
-      }
-    )
+    let config, spec
+    try {
+      [config, spec] = await Promise.all([this.config.exists(), this.spec.exists(), this.version.check(repoVersion)])
+      return config && spec
+    } catch (err) {
+      return false
+    }
   }
 
   /**
    * Close the repo and cleanup.
    *
-   * @param {function(Error)} callback
-   * @returns {void}
+   * @returns {Promise<void>}
    */
-  close (callback) {
+  async close () {
     if (this.closed) {
-      return callback(new Error('repo is already closed'))
+      throw new Error('repo is already closed')
     }
-
     log('closing at: %s', this.path)
-    series([
-      (cb) => this.apiAddr.delete(ignoringNotFound(cb)),
-      (cb) => {
-        each(
-          [this.blocks, this.keys, this.datastore],
-          (store, callback) => store.close(callback),
-          cb)
-      },
-      (cb) => {
-        log('unlocking')
-        this.closed = true
-        this._closeLock(cb)
-      },
-      (cb) => {
-        this.lockfile = null
-        cb()
-      }
-    ], (err) => callback(err))
+    await this.apiAddr.delete()
```

Comment:
This refactored part will fail if the /api does not exists; see original `this.apiAddr.delete(ignoringNotFound(cb))`.